Sources/ScipioKit/Producer/PIF/XCBuildExecutor.swift

```diff
@@ -0,0 +1,231 @@
+        switch info.result {
+        case .ok: log("build completed")
+        case .failed: log(level: .error, "build failed")
+        case .cancelled: log(level: .error, "build cancelled")
+        case .aborted: log(level: .error, "build aborted")
+        }
+    }
+
+    private func handleTargetStarted(info: XCBuildMessage.TargetStartedInfo) {
+        targets[info.targetID] = info
+        log(target: info.targetName, "started")
+    }
+
+    private func handleTargetDiagnostic(info: XCBuildMessage.TargetDiagnosticInfo) {
+        guard let target = targets[info.targetID] else { return }
+        log(target: target.targetName, info.message)
+    }
+
+    private func handleTargetComplete(info: XCBuildMessage.TargetCompleteInfo) {
+        guard let target = targets[info.targetID] else { return }
+        log(target: target.targetName, "completed")
+    }
+
+    private func handleTaskStarted(info: XCBuildMessage.TaskStartedInfo) {
+        tasks[info.taskID] = info
+        let target = info.targetID.flatMap { targets[$0] }
+
+        log(target: target?.targetName, task: info.taskID, info.executionDescription)
+        if let commandLine = info.commandLineDisplayString {
+            log(commandLine)
+        }
+    }
+
+    private func handleTaskDiagnostic(info: XCBuildMessage.TaskDiagnosticInfo) {
+        guard let task = tasks[info.taskID] else { return }
+        let target = task.targetID.flatMap { targets[$0] }?.targetName
+        log(target: target, task: task.taskID, info.message)
+    }
+
+    private func handleTaskOutput(info: XCBuildMessage.TaskOutputInfo) {
+        guard let task = tasks[info.taskID] else { return }
+        let target = task.targetID.flatMap { targets[$0] }?.targetName
+        log(target: target, task: task.taskID, info.data)
+    }
+
+    private func handleTaskComplete(info: XCBuildMessage.TaskCompleteInfo) {
+        guard let task = tasks[info.taskID] else { return }
+        let target = task.targetID.flatMap { targets[$0] }?.targetName
+
+        switch info.result {
+        case .success: break
+        case .failed:
+            log(level: .error, target: target, task: task.taskID, "failed")
+        case .cancelled:
+            log(level: .error, target: target, task: task.taskID, "canceleld")
+        }
+    }
+
+    private func log(
+        level: Logger.Level = .info,
+        target: String? = nil,
+        task: Int? = nil,
+        _ message: String
+    ) {
+        let label: String? = {
+            var str = ""
+            if let target {
+                str += target
+            }
+            if let task {
+                str += "#" + task.description
+            }
+            if str.isEmpty { return nil }
+            return "[" + str + "]"
+        }()
+
+        var str = ""
+        if let label {
+            str += label + " "
+        }
+        str += message
+
+        allMessages.append(str)
```

Comment:
こんな感じでimmutableにできないですか？

```suggestion
        let outputMessage = [
            label,
            message
        ]
            .compactMap { $0 }
            .joined(separator: " ")
            
        allMessages.append(outputMessage)
```

Reply:
してみました。